opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -292,6 +295,20 @@ impl LogProcessor for BatchLogProcessor {
                 otel_warn!(name: "BatchLogProcessor.LogDroppingStarted",
                     message = "BatchLogProcessor dropped a LogRecord due to queue full/internal errors. No further log will be emitted for further drops until Shutdown. During Shutdown time, a log will be emitted with exact count of total logs dropped.");
             }
+            return;
+        }
+
+        // At this point, sending the log record to the data channel was successful.
+        // Increment the current batch size and check if it has reached the max export batch size.
+        if self.current_batch_size.fetch_add(1, Ordering::Relaxed) >= self.max_export_batch_size {
+            // Check if the a control message for exporting logs is already sent to the worker thread.
+            // If not, send a control message to export logs.
+            // `export_log_message_sent` is set to false ONLY when the worker thread has processed the control message.
+            if !self.export_log_message_sent.swap(true, Ordering::Relaxed) {
```

Comment:
should we use `compare_exchange` instead of `swap`. This will avoid redundant writes if the `export_log_message_sent` is already true?

Reply:
I used `swap` because it's less verbose and `swap` uses `compare_exchange` internally anyway. I pushed some new changes. I have added a new if check using atomic load to avoid aggressively calling `swap`. We would now call `swap` only if the atomic load value returned `false`. Let me know what you think.